iscsi/iscsi.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2017 The Prometheus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iscsi
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/prometheus/procfs/internal/fs"
+)
+
+// iscsi target started with /sys/kernel/config/target/iscsi/iqn*
+// configfs + target/iscsi/iqn*
+// IqnGlob is representing all the possible IQN
+const IqnGlob = "target/iscsi/iqn*"
+
+// TargetCore static path /sys/kernel/config/target/core for node_exporter
+// reading runtime status
+const TargetCore = "target/core"
+
+// FS represents the pseudo-filesystem configfs, which provides an interface to
+// iscsi kernel data structures in /sys/kernel/config.
+type FS struct {
+	configfs *fs.FS
+}
+
+// NewFS returns a new configfs mounted under the given mount point. It will
+// error and return empty FS. If the mount point can't be read. For the ease of 
```

Comment:
Looks like there is a trailing space at the end of this line which is causing a CI failure on the go fmt check.

Reply:
sorry about that, forget to run gofmt before submitting and I was working on adding the test case and rewrite the getLunLinkTarget, and it should be all fix now with the new test case. Thanks for looking into it.